classes/core/APIRouter.inc.php

```diff
@@ -77,7 +80,16 @@ function route($request) {
 		// Ensure slim library is available
 		require_once('lib/pkp/lib/vendor/autoload.php');
 
-		$sourceFile = sprintf('api/%s/%s/index.php', $this->getVersion(), $this->getEntity());
+		$sourceFile = null;
+		$entity = $this->getEntity();
+		$sourceFile = sprintf('api/%s/%s/index.php', $this->getVersion(), $entity[0]);
+
+		if (!empty($entity[1])) {
```

Comment:
OK. I still think it would be a better idea to support arbitrary levels of depth, rather than just two. I think the code would be cleaner written that way (and no extra complexity, I don't think), and it'll save a developer confusion if they expect to be able to nest them as deep as they want, which would be a reasonable expectation, I think. APIRouter is a framework concern, and shouldn't have to care whether the business case is going to limit itself to one or two levels of depth.